client/App.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+import { Routes, Route } from 'react-router-dom';
+import SubNavbar from './components/Sub-Navbar';
+
+import Navbar from './components/Navbar';
+
+import Home from './pages/Home';
+import Character from './pages/Character';
+
+function App() {
+  const [characterData, setCharacterData] = useState(undefined);
+  const [subNavbarText, setSubNavBarText] = useState('SEARCH');
+
+  return (
+    <div className="App">
+      <Navbar />
+      <div className="mt-14">
+        <SubNavbar subNavbarText={subNavbarText} setSubNavBarText={setSubNavBarText} />
```

Comment:
So it looks like 3 components can modify the subnavbar text? (SubNavbar, Home, and Character)

If I look at SubNavbar, I don't see that it actually accepts a prop named `setSubNavBarText`. And if it did, it would be weird -- why would the subnavbar be setting its own text?

I finally figured out what you are doing with Home and Character. You are modifying the subnavbar text whenever those two come into view (that is, when they are mounted). But they are mounted when the Route changes.

It seems like a better design would be to calculate the subnavbar text here in App based on the Route. I think you can get the current Route from ReactRouter. If so, you could get rid of the `subNavbarText` state and just calculate it instead.

This is optional. Move forward and come back to this if you need to.

Reply:
I'll keep this in mind and come back to it at the end.